include/libvfio-user.h

```diff
@@ -66,7 +66,7 @@ typedef void *vfu_dma_addr_t;
 typedef struct {
     vfu_dma_addr_t dma_addr;
     int region;
-    int length;
+    uint64_t length;
```

Comment:
but dma_init_sg(), and various friends, still take a 32-bit value, so I'm not sure how this works.

Also, we'd need to do something about vfu_dma_read / vfu_dma_write() - right now they are truncating a potentially 64-bit sg->length

Reply:
Shouldn't it be safe to put 32bit(unsigned) value into 64bit(unsigned)?
Or should we also change dma_init_sg() and friends to receive length in uint64_t

For vfu_dma_read / vfu_dma_write() right now the protocol expects 'count' to be uint32_t. How about having value check for <UINT32_MAX?

Reply:
Was the original bug from using int rather uint32_t ?
You made a change such that dma_controller_dirty_page_get() takes a uint64_t, but without changing dma_init_sg() etc. you're dropping the top 32 bits on the floor